framework/src/main/java/io/github/kgress/scaffold/BaseComponent.java

```diff
@@ -0,0 +1,234 @@
+             */
+            listOfComponents.add(componentInstance);
+          } catch (InstantiationException | IllegalAccessException | InvocationTargetException |
+              NoSuchMethodException e) {
+            throw new ComponentException(e);
+          }
+        });
+    return listOfComponents;
+  }
+
+  /**
+   * Converts {@link Field}'s from a class that extends off of {@link BaseComponent} from an
+   * "inaccessible" state to "accessible." We will only modify the access of Scaffold strongly typed
+   * elements. Afterwards, combines the parent and child together.
+   *
+   * @param componentInstance  the instance of the {@link BaseComponent}
+   * @param fullParentSelector the parent selector being used as the prefix
+   * @param <T>                the type reference of {@link BaseComponent}
+   */
+  private <T extends BaseComponent> void convertFieldsWithNewLocator(T componentInstance,
+      String fullParentSelector) {
+    var classFields = componentInstance.getClass().getDeclaredFields();
+
+    Arrays.stream(classFields).forEach(field -> {
+      var elementType = field.getGenericType();
+      try {
+        field.setAccessible(true);
+        BaseWebElement convertedElement = null;
+        if (elementType.getTypeName().contains("DivWebElement")) {
+          convertedElement = (DivWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("ButtonWebElement")) {
+          convertedElement = (ButtonWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("CheckBoxWebElement")) {
+          convertedElement = (CheckBoxWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("DateWebElement")) {
+          convertedElement = (DateWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("DropDownWebElement")) {
+          convertedElement = (DropDownWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("ImageWebElement")) {
+          convertedElement = (ImageWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("InputWebElement")) {
+          convertedElement = (InputWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("LinkWebElement")) {
+          convertedElement = (LinkWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("RadioWebElement")) {
+          convertedElement = (RadioWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("StaticTextWebElement")) {
+          convertedElement = (StaticTextWebElement) field.get(componentInstance);
+        } else if (elementType.getTypeName().contains("BaseWebElement")) {
+          convertedElement = (BaseWebElement) field.get(componentInstance);
```

Comment:
Here's a neat thing to try out that would be a little cleaner.

You can have a list of class names you want to support (could even build that dynamically if you'd like, any class that is a child of BaseWebElement):

```java

// let's say you defined your list of string class names here (manually at first)
String[] listOfScaffoldClasses = { ... };

// filter out for just the class name that matches the one you have
Optional<String> classNameEquivalent = Arrays.stream(classNames)
          .filter(clazz -> elementType.getTypeName().contains(clazz))
          .findFirst();

// if it was not there, log like you were before
  if (classNameEquivalent.isEmpty()){
    log.debug(
              "Scaffold detected a field during component list building that is not a defined as "
This conversation was marked as resolved by kgress
                  + "a strongly typed element. Skipping modification of field.");
  }

  // do a dynamic cast
  try{
      Class.forName(classNameEquivalent.get()).cast(field.get(componentInstance));
  }catch (ClassCastException e){
      // should only happen if your original list of types was invalid
  }

If the original list was also generated at runtime and cached, would allow scaffold to adjust dynamically for new strong types when they're added/removed.

Reply:
Great suggestion overall. Enjoyed our conversation on this. End result:
```
var classFields = componentInstance.getClass().getDeclaredFields();

    Arrays.stream(classFields).forEach(field -> {
      try {
        field.setAccessible(true);
        if (field.get(componentInstance) instanceof BaseWebElement) {
          convertField(componentInstance, (BaseWebElement) field.get(componentInstance), field,
              fullParentSelector);
        } else {
          log.debug(String.format(
              "Scaffold detected the field [%s] during component list building that "
              + "is not a defined as a strongly typed element. Skipping conversion of field.",
              field));
        }
        field.setAccessible(false);
      } catch (InvocationTargetException | IllegalAccessException | NoSuchMethodException |
          InstantiationException e) {
        throw new ComponentException(e);
      }
    });
```